msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByInteractiveFlowSupplier.java

```diff
@@ -209,7 +209,7 @@ private AuthorizationResult getAuthorizationResultFromHttpListener() {
                 expirationTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()) + 1;
             }
 
-            while (result == null && !interactiveRequest.futureReference().get().isCancelled()) {
+            while (result == null && !interactiveRequest.futureReference().get().isDone()) {
```

Comment:
Timeout in interactive flow is not something MSALs do and it's not something we ask app dev to do either. It can take a really long time to perform auth - e.g. I was just asked to MFA and then to change password.

But anyway, it looks like this is based on public API, so it's fine.

Reply:
For interactive flow we have a timeout parameter that would cause this loop to end after a default 120 seconds: https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/ebdd4fc7299fdf31b2534c25e18da5b036b84292/msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/InteractiveRequestParameters.java#L100

This ended up being kind of a weird situation. In Java 8 there wasn't really a timeout option for ending a future (there was [a timeout](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#get-long-java.util.concurrent.TimeUnit-) for waiting for the result), however in Java 9 they started introducing [methods](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/util/concurrent/CompletableFuture.html#completeOnTimeout-T-long-java.util.concurrent.TimeUnit-) for completing the future after a timeout. Since we target Java 8 as a minimum but also want to support newer versions, this means
- Java 8 customers using that `httpPollingTimeoutInSeconds` as a timeout to end the interactive flow
- Java 9+ customers using the newer timeout methods in Java's CompletableFuture API and expecting MSAL to follow it